ext/pdo_odbc/odbc_driver.c

```diff
@@ -375,18 +375,13 @@ static int odbc_handle_get_attr(pdo_dbh_t *dbh, zend_long attr, zval *val)
 static zend_result odbc_handle_check_liveness(pdo_dbh_t *dbh)
 {
 	RETCODE ret;
-	UCHAR d_name[32];
-	SQLSMALLINT len;
+	SQLUINTEGER dead;
 	pdo_odbc_db_handle *H = (pdo_odbc_db_handle *)dbh->driver_data;
 
-	/*
-	 *  XXX: Should we be using SQL_ATTR_CONNECTION_DEAD? Procedural ODBC
-	 *  uses this check, but it might be problematic per #20298
-	 */
-	ret = SQLGetInfo(H->dbc, SQL_DATA_SOURCE_READ_ONLY, d_name,
-		sizeof(d_name), &len);
+	/* ODBC 3.5; procedural ODBC uses SQLGetInfo read only instead */
```

Comment:
Oh, you're right, `SQL_ATTR_CONNECTION_DEAD` is ODBC 3.5. We may need a fallback for ODBC 3.0 (maybe just not checking the liveliness there is okay).

Reply:
Yeah, I'm not having much luck with the Db2i driver on Linux; it reconnects, but dead returns both SQL_SUCCESS and dead == 0 (SQL_CD_FALSE). (And a segfault at the end, but I can only assume that might be my fault for running bleeding-edge Fedora.) I can try it on i directly too, but it's the same codebase on both.

If drivers can't report their dead status reliably, maybe using SQLGetInfo only or to verify if the driver is correct would work.